src/python/WMComponent/DBS3Buffer/MySQL/CheckDrainStatus.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import print_function, division
+from WMCore.Database.DBFormatter import DBFormatter
+
+class CheckDrainStatus(DBFormatter):
+    """
+    _CheckDrainStatus_
+    
+    1. checks all the workflows are completed status (within the agent)
+    2. checks all the blocks are closed
+    3. checks all the files are injected to PhEDEx
+    4. checks all the files are uploaded to DBS
+
+    TODO: need to check archived statsus
+    """
+    completedSQL = "SELECT count(*) FROM dbsbuffer_workflow WHERE completed = 0"
+    
+    blockClosedSQL = "SELECT count(*) FROM dbsbuffer_block where status != 'Closed'"
+
+    phedexSQL = "SELECT count(*) FROM dbsbuffer_file WHERE in_phedex = 0"
+    
+    dbsSQL = "SELECT count(*) FROM dbsbuffer_file WHERE status != 'InDBS'"
```

Comment:
In theory, any files should be injected into phedex. I put it in my script because I saw such cases in the past and we shouldn't bother about it.
Giving it a second though - this code will be in WMCore, not a dummy script :) - maybe we don't filter anything on this query, but try to track down those cases and make sure this table is in a clean and consistent state. What do you think?